src/FlatSharp.Runtime/IO/InputBufferExtensions.cs

```diff
@@ -138,6 +139,23 @@ private static void ThrowInvalidVtableException()
             return buffer.GetReadOnlyMemory().Slice(uoffset + sizeof(uint), (int)buffer.ReadUInt(uoffset));
         }
     }
+    
+    public static Span<TElement> UnsafeReadSpan<TBuffer, TElement>(this TBuffer buffer, int uoffset) where TBuffer : IInputBuffer where TElement : struct
+    {
+        checked
+        {   
+            // The local value stores a uoffset_t, so follow that now.
+            uoffset = uoffset + buffer.ReadUOffset(uoffset);
+
+            // We need to construct a Span<TElement> from byte buffer that:
+            // 1. starts at correct offset for vector data
+            // 2. has a length based on *TElement* count not *byte* count
+            var byteSpanAtDataOffset = buffer.GetSpan().Slice(uoffset + sizeof(uint));
```

Comment:
One note on `.GetSpan()`, which is that `ReadOnlyMemoryInputBuffer` will throw for this case. Since NativeArray is mutable, this seems appropriate to me, but just wanted to call it out.